pkg/storage/filepath/version.go

```diff
@@ -0,0 +1,57 @@
+package filepath
+
+import (
+	"fmt"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func getResourceVersion(obj runtime.Object) (uint64, error) {
+	if obj == nil {
```

Comment:
(FWIW `meta.Accessor()` will return an error if passed nil, so won't panic)

There is create-on-update logic where you end up getting with a `nil` object in the update paths, so this was convenience for that.

Sidenote: In the K8s/etcd code, there's a second arg for the `UpdateFunc` you pass in which is `ResponseMeta` that includes the revision + TTL from the server, so they use that instead of reading from the obj within the fn - maybe that's better? 🤷

Reply:
(I can also explicitly handle the 2 `nil` special cases if you think this is a footgun)